libs/evcs-service/src/main/java/uk/gov/di/ipv/core/library/client/EvcsClient.java

```diff
@@ -140,9 +148,12 @@ public void updateUserVCs(String userId, List<EvcsUpdateUserVCsDto> evcsUserVCsT
 
     private URI getUri(String userId, List<EvcsVCState> vcStatesToQueryFor)
             throws URISyntaxException {
-        var uriBuilder =
-                new URIBuilder(configService.getSsmParameter(EVCS_APPLICATION_URL))
-                        .setPathSegments("vcs", userId);
+        URI evcsUri = new URI(configService.getSsmParameter(EVCS_APPLICATION_URL));
+        List<String> pathSegments =
+                evcsUri.getPath() != null
+                        ? List.of(evcsUri.getPath().replaceFirst("/", ""), "vcs", userId)
```

Comment:
Will this work if you have more than one path segment?

I know I asked you not to do this in the first place, but if we're having to break this down anyway, then it's probably easier just to construct it properly ourselves, i.e.:
```
var baseUri = "%s/vcs/%s".formatted(
        configService.getSsmParameter(EVCS_APPLICATION_URL),
        URLEncoder.encode(userId, StandardCharsets.UTF_8));

var uriBuilder = new URIBuilder(baseUri);

...
```

Reply:
Done and added test.